src/shims/env.rs

```diff
@@ -109,3 +109,10 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 }
+
+pub(crate) fn get_pid<'tcx>(ecx: &InterpCx<'tcx, MiriMachine<'tcx>>, item_name: &str) -> u32 {
+    match ecx.check_no_isolation(item_name) {
```

Comment:
We never ever match on `InterpResult`. Instead of this, just check `machine.communicate()` directly.

Reply:
done.

Could you explain a bit why we should never match `InterpResult`?

Reply:
InterpResult is constructed when we want to abort execution, e.g. on UB. It is always a bug to do anything but abort execution at that point.

Reply:
I wish we had a way to ensure that `InterpResult` is never "caught" in this way...

Reply:
I have never done this, so I don't know how easy it is, but can't some lint be created for that?

Reply:
Yeah, but clippy would have to know which error types are okay with a `Result<T, E>` to be `match`ed on and which not. 

Anyway, that's off-topic here.